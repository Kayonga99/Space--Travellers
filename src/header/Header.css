.hr {
  width: 90%;
  background-color: black;
  align-self: center;
  margin: 0 auto;
}

.brand {
  display: inline-flex;
  gap: 0.5em;
}

.navbar-nav {
  gap: 1em;
}
