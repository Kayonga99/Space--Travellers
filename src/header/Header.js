import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import React from 'react';
import './Header.css';

const Header = () => (
  <div>
    <Navbar bg="white" expand="lg">
      <Container className="header">
        <Navbar.Brand className="brand" href="#home">
          <img src="/assets/rocket-placeholder.png" width="30" height="30" alt="logo" />
          <h2> Space Travelers&rsquo;Hub</h2>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav" className="justify-content-end">
          <Nav>
            <Nav.Link href="/rockets">Rockets</Nav.Link>
            <Nav.Link href="/missions">Missions</Nav.Link>
            <div className="line" />
            <Nav.Link href="/profile">My Profile</Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
    <hr className="hr" />
  </div>
);

export default Header;
