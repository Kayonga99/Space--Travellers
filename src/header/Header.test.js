import { render, screen, fireEvent } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import Header from './Header';
import '@testing-library/jest-dom';

expect.extend({ toMatchDiffSnapshot });

test('@Header Rockets Navigation', () => {
  const { asFragment } = render(<HashRouter><Header /></HashRouter>);
  fireEvent.click(screen.getByText(/Rockets/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Rockets/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

test('@Header Missions Navigation', () => {
  const { asFragment } = render(<HashRouter><Header /></HashRouter>);
  fireEvent.click(screen.getByText(/Missions/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Missions/));
  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

test('@Header My Profile Navigation', () => {
  const { asFragment } = render(<HashRouter><Header /></HashRouter>);
  fireEvent.click(screen.getByText(/My Profile/));
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/My Profile/));
  expect(firstRender).toMatchDiffSnapshot(asFragment());
});
