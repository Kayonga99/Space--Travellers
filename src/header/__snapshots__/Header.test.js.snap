// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Header Mission Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header Missions Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header My Profile Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header Profile Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header Rocket Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`@Header Rockets Navigation 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
